### ISSUE TITLE
can't import external http to https redirecting images by default csv import

### ISSUE THREAD

**Issue by ankit-sf** [magento/magento2#18839]
### Preconditions (*)

1. Magento 2.1.7
2. or any later version

### Steps to reproduce (*)

1. use csv import for products, use http image with redirection to https with 302 status.
ex. http://midwest-wholesale-staging.cloud.akeneo.com/bundles/pimui/images/info-user.png
2. it shows error

> Imported resource (image) could not be downloaded from external 
> resource due to timeout or access permissions

### Expected result (*)

1. Image should be imported in magento2 and added to product

### Actual result (*)

1. Image not gets imported in magento. 
http://i.prntscr.com/dc3bhWu_SGq1-whwP25AdQ.png
Also shows error
> Imported resource (image) could not be downloaded from external 
> resource due to timeout or access permissions

**Comment by magento-engcom-team** [comment]
Hi @ankit-sf. Thank you for your report. 
To help us process this issue please make sure that you provided the following information:
- [x] Summary of the issue
- [x] Information on your environment
- [x] Steps to reproduce
- [x] Expected and actual results

Please make sure that the issue is reproducible on the vanilla Magento instance following **Steps to reproduce**. To deploy vanilla Magento instance on our environment, please, add a comment to the issue: 

`@magento-engcom-team give me $VERSION instance` 

where `$VERSION` is [version tags](https://github.com/magento/magento2/tags) (starting from 2.2.0+) or develop branches (for example: 2.3-develop). 
For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation.

@ankit-sf do you confirm that you was able to reproduce the issue on vanilla Magento instance following steps to reproduce?
- [x] yes
- [ ] no

**Comment by ankit-sf** [comment]
@magento-engcom-team give me 2.1.7 instance

**Comment by magento-engcom-team** [comment]
Hi @ankit-sf. Thank you for your request.
Unfortunately, I can only deploy instances for `2.2` and `2.3` release lines

**Comment by ankit-sf** [comment]
@magento-engcom-team give me 2.2.3 instance

**Comment by magento-engcom-team** [comment]
Hi @ankit-sf. Thank you for your request. I'm working on Magento 2.2.3 instance for you

**Comment by magento-engcom-team** [comment]
Hi @ankit-sf, here is your [Magento instance](http://ec2-34-228-235-121.compute-1.amazonaws.com/i-18839-2-2-3/).
Admin access: [http://ec2-34-228-235-121.compute-1.amazonaws.com/i-18839-2-2-3//admin](http://ec2-34-228-235-121.compute-1.amazonaws.com/i-18839-2-2-3//admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by ankit-sf** [comment]
yes issue replicated with following csv data:

```
sku,store_view_code,attribute_set_code,product_type,categories,product_websites,name,description,short_description,weight,product_online,tax_class_name,visibility,price,special_price,special_price_from_date,special_price_to_date,url_key,meta_title,meta_keywords,meta_description,base_image,base_image_label,small_image,small_image_label,thumbnail_image,thumbnail_image_label,swatch_image,swatch_image_label,created_at,updated_at,new_from_date,new_to_date,display_product_options_in,map_price,msrp_price,map_enabled,gift_message_available,custom_design,custom_design_from,custom_design_to,custom_layout_update,page_layout,product_options_container,msrp_display_actual_price_type,country_of_manufacture,additional_attributes,qty,out_of_stock_qty,use_config_min_qty,is_qty_decimal,allow_backorders,use_config_backorders,min_cart_qty,use_config_min_sale_qty,max_cart_qty,use_config_max_sale_qty,is_in_stock,notify_on_stock_below,use_config_notify_stock_qty,manage_stock,use_config_manage_stock,use_config_qty_increments,qty_increments,use_config_enable_qty_inc,enable_qty_increments,is_decimal_divided,website_id,related_skus,related_position,crosssell_skus,crosssell_position,upsell_skus,upsell_position,additional_images,additional_image_labels,hide_from_product_page,custom_options,bundle_price_type,bundle_sku_type,bundle_price_view,bundle_weight_type,bundle_values,bundle_shipment_type,configurable_variations,configurable_variation_labels,associated_skus
Simple,,Default,simple,,base,Simple,,,,1,"Taxable Goods","Catalog, Search",1.0000,,,,simple,Simple,Simple,"Simple ",http://midwest-wholesale-staging.cloud.akeneo.com/bundles/pimui/images/info-user.png,,http://midwest-wholesale-staging.cloud.akeneo.com/bundles/pimui/images/info-user.png,,http://midwest-wholesale-staging.cloud.akeneo.com/bundles/pimui/images/info-user.png,,http://midwest-wholesale-staging.cloud.akeneo.com/bundles/pimui/images/info-user.png,,"10/26/18, 9:38 AM","10/26/18, 9:40 AM",,,"Block after Info Column",,,,"Use config",,,,,,,,,,0.0000,0.0000,1,0,0,1,1.0000,1,10000.0000,1,0,1.0000,1,1,1,1,1.0000,1,0,0,0,,,,,,,http://midwest-wholesale-staging.cloud.akeneo.com/bundles/pimui/images/info-user.png,,,,,,,,,,,,

```

**Comment by magento-engcom-team** [comment]
Hi @rahulwebkul. Thank you for working on this issue. 
In order to make sure that issue has enough information and ready for development, please read and check the following instruction: :point_down:
- [x] 1. Verify that issue has all the required information. (**Preconditions**, **Steps to reproduce**, **Expected result**, **Actual result**).<details><summary>Details</summary>If the issue has a valid description, the label `Issue: Format is valid` will be added to the issue automatically. Please, edit issue description if needed, until label `Issue: Format is valid` appears.</details>

- [x] 2. Verify that issue has a meaningful description and provides enough information to reproduce the issue. If the report is valid, add `Issue: Clear Description` label to the issue by yourself.

- [x] 3. Add `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.
- [x] 4. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [x] 5. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [x] 6. Add label `Issue: Confirmed` once verification is complete. 

- [x] 7. Make sure that automatic system confirms that report has been added to the backlog.

**Comment by ghost** [comment]
#SQUASHTOBERFEST

**Comment by magento-engcom-team** [comment]
@rahulwebkul Thank you for verifying the issue.

Unfortunately, not enough information was provided to created internal ticket. Please consider adding the following:
- [x] Add `"Reproduced on "` label(s) to this ticket based on verification result

Once all required information is added, please add label `"Issue: Confirmed"` again.
Thanks!

**Comment by magento-engcom-team** [comment]
@rahulwebkul Thank you for verifying the issue. Based on the provided information internal tickets `MAGETWO-95956`, `MAGETWO-95957` were created

**Comment by sidolov** [comment]
Hi @ankit-sf. Thank you for your report. 
The issue has been fixed in magento/magento2#18900 by @rahulwebkul in 2.3-develop branch
Related commit(s):
 - [22abbcbb6ac21b32ca25b93f55bba8079a2d56c9](https://github.com/magento/magento2/commit/22abbcbb6ac21b32ca25b93f55bba8079a2d56c9)
 - [8aafa54f3356a8d361231cf3d84e19f80ef076c0](https://github.com/magento/magento2/commit/8aafa54f3356a8d361231cf3d84e19f80ef076c0)

The fix will be available with the upcoming 2.3.1 release.

**Comment by magento-engcom-team** [comment]
Hi @ankit-sf. Thank you for your report. 
The issue has been fixed in magento/magento2#18899 by @rahulwebkul in 2.2-develop branch
Related commit(s):
 - [3a88a38399cf011c737f067be9ec5cb60f99fbd9](https://github.com/magento/magento2/commit/3a88a38399cf011c737f067be9ec5cb60f99fbd9)

The fix will be available with the upcoming 2.2.8 release.
